extension/encoding/googlecloudlogentryencodingextension/internal/auditlog/parser.go

```diff
@@ -258,14 +312,65 @@ func handlePolicyViolationInfo(info *policyViolationInfo, attr pcommon.Map) {
 	}
 }
 
-func handleRequestMetadata(metadata *requestMetadata, attr pcommon.Map) {
+func handleRequestMetadata(metadata *requestMetadata, attr pcommon.Map) error {
 	if metadata == nil {
-		return
+		return nil
 	}
 
 	shared.PutStr(string(semconv.ClientAddressKey), metadata.CallerIP, attr)
 	shared.PutStr(string(semconv.UserAgentOriginalKey), metadata.CallerSuppliedUserAgent, attr)
 	shared.PutStr(gcpAuditRequestCallerNetwork, metadata.CallerNetwork, attr)
+
+	if metadata.RequestAttributes != nil {
+		if err := shared.AddStrAsInt(string(semconv.HTTPRequestSizeKey), metadata.RequestAttributes.Size, attr); err != nil {
+			return fmt.Errorf("failed to add http request size %s: %w", metadata.RequestAttributes.Size, err)
```

Comment:
Does returning an error from this function mean that payloads that were previously parsed correctly may now fail to be parsed?